src/core/lib/gprpp/arena.cc

```diff
@@ -31,19 +31,36 @@
 #include "src/core/lib/gpr/alloc.h"
 #include "src/core/lib/gprpp/memory.h"
 
+namespace grpc_core {
+
 template <size_t alignment>
-static void* gpr_arena_malloc(size_t size) {
+static void* _aligned_alloc(size_t size) {
+#if defined(GPR_LINUX)
```

Comment:
Is `aligned_alloc()` only available on Linux?  If it could be available on other platforms, then it might be better to add a new `GPR_HAS_ALIGNED_ALLOC` macro to port_platform.h and use that to decide whether to use it.

Reply:
aligned_alloc is available on Linux in c++11 in stdlib.h.
std::aligned_alloc is available in c++17 in <cstdlib>.
It looks like Windows gives _aligned_alloc.
For osx I think the solution is to use posix_memalign.

Then, for all other platforms, for now we can devolve to our implementation.

Reply:
Added as requested.